nsdb-core/src/main/scala/io/radicalbit/nsdb/post_proc/package.scala

```diff
@@ -268,6 +275,15 @@ package object post_proc {
             NSDbNumericType(bits.map(_.value.rawValue.asInstanceOf[Long]).sum),
             foldMapOfBit(bits, bit => bit.dimensions),
             foldMapOfBit(bits, bit => bit.tags))
+
+      case CountDistinctAggregation =>
+        val uniqueValues = bits.foldLeft(Set.empty[NSDbType])((acc, b2) => acc ++ b2.uniqueValues)
+
+        if (finalStep)
+          Bit(0, uniqueValues.size.toLong, Map.empty, bits.head.tags)
```

Comment:
Are we guaranteed `bits.head`does not fail by design?

Reply:
I will reinforce it. Thanks